mindmeld/augmentation.py

```diff
@@ -0,0 +1,285 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2015 Cisco Systems, Inc. and others.  All rights reserved.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""This module contains the data augmentation processes for MindMeld."""
+
+import logging
+import re
+
+from abc import ABC, abstractmethod
+from tqdm import tqdm
+
+from .components._config import get_augmentation_config, get_language_config
+from .constants import _get_pattern
+from .models.helpers import register_augmentor
+from .resource_loader import ResourceLoader
+
+logger = logging.getLogger(__name__)
+
+try:
+    import torch
+except ImportError:
+    logger.info(
+        "Library not found: 'torch'. Run 'pip install mindmeld[augment]'" " to install."
+    )
+
+try:
+    from transformers import PegasusForConditionalGeneration, PegasusTokenizer
+    from transformers import MarianMTModel, MarianTokenizer
+except ImportError:
+    logger.info(
+        "Library not found: 'transformers'. Run 'pip install mindmeld[augment]'"
+        " to install."
+    )
```

Comment:
It would be cleaner to add this code block for trying to import torch into a `register_dependecies` function and follow the pattern in `mindmeld/path.py::607`